packages/davinci-client/src/lib/node.utils.ts

```diff
@@ -0,0 +1,35 @@
+import { DavinciErrorResponse } from './davinci.types';
+import { CollectorErrors } from './node.types';
+
+export function getCollectorErrors(error: DavinciErrorResponse) {
+  const details = error.details;
+  if (!details || !Array.isArray(details)) {
+    return [];
+  }
+  return details.reduce((acc, next) => {
+    if (!next.rawResponse) {
+      return acc;
+    }
+    if (!next.rawResponse.code) {
+      return acc;
+    }
+    if (next.rawResponse.code !== 'INVALID_DATA') {
+      return acc;
+    }
+    if (!Array.isArray(next.rawResponse.details)) {
+      return acc;
+    }
+    next.rawResponse.details.forEach((item): void => {
+      if (!item.target) {
+        return;
+      }
+      acc.push({
+        code: item.code || '',
+        message: item.message || '',
+        target: item.target,
+      });
+    });
+
+    return acc;
+  }, [] as CollectorErrors[]);
```

Comment:
Maybe nitpicky but you can remove the `as` cast here


and use the generic on Reduce

```
return details.reduce<CollectorErrors[]>((acc, next) =>...
```